scalyr_agent/builtin_monitors/journald_monitor.py

```diff
@@ -557,16 +557,12 @@ def format_msg(
     ):
         string_buffer = six.StringIO()
 
-        string_buffer.write(
-            "%s %s" % (metric_name, scalyr_util.json_encode(metric_value))
-        )
+        string_buffer.write("%s %s" % (metric_name, metric_value))
```

Comment:
Looking at these changes, a little worried what happens if a customer gets tripped up by these changes.  Unfortunately, I think we need a configuration option that allows them to go back to the original format, just in case.

We could make it configuration option like "format_version" and document this new format as version "2" and the original as "1".  We default to 2 but allow them to switch back.  That applies to this line and then the changes down below to how we emit the extra fields.

Reply:
One difference I noticed here is removing `json_encode` removes the quotes around the message. I would suggest leaving the surrounding quotes, but that brings up the question of what happens to parsing if the message has quotes in it, and if we `\` escape those we just come back to the issue this was all about...